packages/stable/src/api/functions/functionsApi.ts

```diff
@@ -0,0 +1,142 @@
+// Copyright 2020 Cognite AS
+
+import { BaseResourceAPI } from '@cognite/sdk-core';
+import type {
+  CogniteInternalId,
+  IdEither,
+  ItemsWrapper,
+} from '@cognite/sdk-core';
+import type {
+  CogniteFunction,
+  ExternalCogniteFunction,
+  FunctionDeleteParams,
+  FunctionIdEither,
+  FunctionListScope,
+  FunctionsActivationResponse,
+  FunctionsLimits,
+} from './types';
+
+export class FunctionsAPI extends BaseResourceAPI<CogniteFunction> {
+  /**
+   * @hidden
+   */
+  protected getDateProps() {
+    return this.pickDateProps(['items'], ['createdTime']);
+  }
+
+  /**
+   * [Create functions](https://developer.cognite.com/api#tag/Functions/operation/postFunctions)
+   *
+   * ```js
+   * const functions = await client.functions.create([{
+   *   name: 'My awesome function',
+   *   fileId: 5467347282343
+   * }]);
+   * ```
+   */
+  public create = (
+    items: ExternalCogniteFunction[]
+  ): Promise<CogniteFunction[]> => {
+    return super.createEndpoint(items);
+  };
+
+  /**
+   * [List functions](https://developer.cognite.com/api#tag/Functions/operation/listFunctions)
+   *
+   * ```js
+   * const functions = await client.functions.list();
+   * ```
+   */
+  public list = async (
+    scope?: FunctionListScope
+  ): Promise<CogniteFunction[]> => {
+    const response = await this.post<ItemsWrapper<CogniteFunction[]>>(
+      this.listPostUrl,
+      { data: scope || {} }
+    );
+    return this.addToMapAndReturn(response.data.items, response);
+  };
+
+  /**
+   * [Retrieve a function by its ID](https://developer.cognite.com/api#tag/Functions/operation/getFunction)
+   *
+   * ```js
+   * const func = await client.functions.retrieve(123);
+   * ```
+   */
+  public retrieve = async (id: CogniteInternalId): Promise<CogniteFunction> => {
+    const response = await this.get<CogniteFunction>(this.url(`${id}`));
+    return this.addToMapAndReturn(response.data, response);
+  };
+
+  /**
+   * [Retrieve functions by IDs](https://developer.cognite.com/api#tag/Functions/operation/byIdsFunctions)
+   *
+   * ```js
+   * const functions = await client.functions.retrieveMultiple([{id: 123}, {externalId: 'abc'}]);
+   * ```
+   */
+  public retrieveMultiple = (ids: IdEither[]): Promise<CogniteFunction[]> => {
+    return super.retrieveEndpoint(ids);
+  };
```

Comment:
maybe enough having the regular array retrieve only? array of one support one element anyway

Reply:
Done